native/src/routes/Events.tsx

```diff
@@ -81,8 +92,68 @@ const Events = ({
       events.find(it => it.slug === slug) ?? events.find(it => it.slug.substring(0, it.slug.indexOf('$')) === slug)
 
     if (event) {
+      const saveEventToCalendar = async (calendarId: string | undefined): Promise<void> => {
+        await RNCalendarEvents.saveEvent(event.title, {
+          startDate: event.date.startDate.toISOString(),
+          endDate: event.date.endDate.toISOString(),
+          allDay: event.date.allDay,
+          location: event.location?.fullAddress,
+          description: event.excerpt,
+          calendarId,
+        })
+          .then(id => {
+            setEventExported(true)
+            showSnackbar({
+              text: t('added'),
+              positiveAction: {
+                label: t('goToCalendar'),
+                onPress: async () => {
+                  if (Platform.OS === 'ios') {
+                    const appleRefDate = moment('Jan 1 2001', 'MMM DD YYYY')
+                    const secondsSinceRefDate = event.date.startDate.diff(appleRefDate, 'seconds')
+                    Linking.openURL(`calshow:${secondsSinceRefDate}`)
+                  } else if (Platform.OS === 'android') {
+                    Linking.openURL(`content://com.android.calendar/events/${id}`)
+                  }
+                },
+              },
+            })
+          })
+          .catch(() => {
+            showSnackbar({ text: 'generalError' })
+          })
+      }
+
+      const handlePress = async (): Promise<void> => {
+        let permission = await RNCalendarEvents.checkPermissions()
+        if (permission === 'undetermined') {
+          permission = await RNCalendarEvents.requestPermissions()
+        }
+        if (permission === 'denied' || permission === 'restricted') {
+          showSnackbar({ text: 'noCalendarPermission' })
+          return
+        }
+        const editableCalendars = (await RNCalendarEvents.findCalendars()).filter(cal => cal.allowsModifications)
+        if (editableCalendars.length === 0) {
+          showSnackbar({ text: 'generalError' })
```

Comment:
i think you should throw an error message like. "No calendars available", not just a general error